src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -1677,6 +1691,511 @@ module T = struct
           let res = try_with_coinbase () in
           make_diff res None
 
+  module Partial_diff = struct
+    type t =
+      { available_slots: int
+      ; paid_for_available_slots: int
+      ; receiver_fee_transfer: bool
+      ; coinbase_fee_transfer:
+          (Amount.t * (Fee.t * Public_key.Compressed.t) option) option
+      ; commands_rev: User_command.Valid.t With_status.t list
+      ; purchased_work_rev: Transaction_snark_work.Checked.t list
+      ; available_fee_slots: int
+      ; fee_transfers: Fee.t Public_key.Compressed.Map.t
+      ; profit: Amount.Signed.t
+            (* We allow this to become negative, since it may rebound with some
+               subsequent low-fee work and high-fee transactions.
+            *)
+      ; prev_best_diff: t option }
+
+    let empty ~receiver ~max_slots ~max_work () =
+      { available_slots= max_slots
+      ; paid_for_available_slots= max_slots - max_work
+      ; receiver_fee_transfer= false
+      ; coinbase_fee_transfer= None
+      ; commands_rev= []
+      ; purchased_work_rev= []
+      ; available_fee_slots= 0
+      ; fee_transfers=
+          (* We pretend that we've already seen a zero fee payment to the
+             [receiver], since any fee payments to it can be bundled into the
+             coinbase, and so we don't need to pay for slots for them.
```

Comment:
TODO: This is a lie, although the behaviour is correct due to `receiver_fee_transfer`.